### PR TITLE
Admin route (soft) view access

### PR DESCRIPTION
**Pull request by carletex**
Let's soft-protect the /admin route. Data is not secret anyway, but let's now allow /admin direct access.

As we discussed on https://github.com/BuidlGuidl/grants.buidlguidl.com/pull/18 and https://github.com/BuidlGuidl/grants.buidlguidl.com/pull/17, if we wanted full protection, we could do SIWE + JWT or signed reads. We might want to consider it in the future.

### CODE REVIEW

**Review by technophile-04 — APPROVED** [review]
Nice !! Also what if maybe add : 

```ts
  useEffect(() => {
    if (!address) router.push("/");
  }, [address, router]);
```

But please feel free to merge it as it 🙌

File: packages/nextjs/app/admin/page.tsx [hunk 1]
```diff
@@ -5,28 +5,41 @@ import { BatchActionModal } from "./_components/BatchActionModal";
...
 import { GrantDataWithBuilder } from "~~/services/database/schema";
 import { PROPOSAL_STATUS } from "~~/utils/grants";
 import { notification } from "~~/utils/scaffold-eth";
 
+const fetcherWithHeader = (url: string, address: string) =>
+  fetch(url, {
+    headers: {
+      Address: address,
+    },
+  }).then(res => res.json());
```
Comment (technophile-04) [inline packages/nextjs/app/admin/page.tsx:19]: To make https://github.com/BuidlGuidl/grants.buidlguidl.com/pull/96/files#r1539245051 work, we need to handle `res.ok` case example : 

```ts
const fetcherWithHeader = async (url: string, address: string) => {
  const res = await fetch(url, {
    method: "GET",
    headers: {
      Address: address,
    },
  });
  const data = await res.json();
  // check if server send 401 or any other error
  if (!res.ok) {
    throw new Error(data.error || `Error getting data`);
  }
  return data;
};
```

**Comment by carletex** [comment]
```
  useEffect(() => {
    if (!address) router.push("/");
  }, [address, router]);
```

Hey @technophile-04 , the problem with this is that `address` might be undefined while the hook is "connecting". It won't happen if you are already on the page and navigate to /admin, but it happens if you go directly to /admin... so you get redirected haha 

image

I'll be interested to know how to fix this for other occasions (`isConnecting` doesn't seem to work)

---

Merging this for now... since I think we'll be changing it soon haha (Austin is more inclined to do a SIWE + validation). I'll open an issue.

**Comment by technophile-04** [comment]
>  but it happens if you go directly to /admin... so you get redirected haha

Ohhhhh yes!! my bad, missed it completely sorry :( 

> (isConnecting doesn't seem to work)

Yeah :(, I think since we are using our custom solution `useAutoConnect` hook for auto connecting I think because of this wagmi isn't aware about its status. 

Umm not sure what's the best solution is maybe very hacky solution : exposing some kindof `isLoading/isConnecting` state from `useAutoConnect` hook 

But yup this should be solved in Wagm v2 though 🙌

**Pull request merged**